### ISSUE TITLE
Questions about EmbeddingWrapper

### ISSUE THREAD

**Issue by irowberry** [lucidrains/perfusion-pytorch#7]
Is the EmbeddingWrapper used to overwrite/train the text encoder's embedding layer? 
From what I understand I just need to access the Embedding layer in let's say CLIP, wrap it in this class, pass in my text, get `embeds_with_new_concept, embeds_with_superclass, embed_mask, concept_indices` out, then finish passing the first two into the rest of the text encoding model. Is it the only layer within the text encoder that needs to be trained? Thanks

**Comment by lucidrains** [comment]
Yup, you can see all the parameters that are trained in this function. This is the highlight of this paper https://github.com/lucidrains/perfusion-pytorch/blob/main/perfusion_pytorch/save_load.py#L20

**Comment by lucidrains** [comment]
@irowberry I added another keyword argument to the embedding wrapper earlier, if you are able to extract the transformer text encoder from clip

**Comment by irowberry** [comment]
Another question, does the embed_mask get passed into `loss_fn_weighted_by_mask()` as well as the cross attention modules?

**Comment by lucidrains** [comment]
@irowberry so `embed_mask` goes into the cross attention as the key padding mask

the mask for weighting the loss function actually comes from another section in the paper. they use some pretrained model that outputs a segmentation mask to weigh the loss, and this apparently led to better results. i would say leave that for the end, as it should work without it
